handycon.py

```diff
@@ -55,19 +56,20 @@
         "QAM": EVENT_QAM,
         "SCR": EVENT_SCR,
     }
-# Capture the username and home path of the first user to log in. This assumes
-# the first user is the only user.
+
+HIDE_PATH = Path(HIDE_PATH)
+# Capture the username and home path of the user who has been logged in the longest.
 USER = None
+HOME_PATH = Path('/home')
 cmd = "who | awk '{print $1}' | sort | head -1"
-while USER == None:
-    USER_LIST = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout = subprocess.PIPE, stderr = subprocess.PIPE, shell=True)
-    for get_first in USER_LIST.stdout:
-        name = get_first.decode().strip()
-        if name is not None:
-            USER = name
-        break
+while USER is None:
+    who = [w.split() for w in os.popen('who').read().strip().split('\n')]
```

Comment:
I receive the following error when running this

```
Traceback (most recent call last):
  File "/home/gamer/forks/HandyGCCS/./handycon.py", line 67, in <module>
    who.sort(key=lambda row: (row[2], row[3]))
  File "/home/gamer/forks/HandyGCCS/./handycon.py", line 67, in <lambda>
    who.sort(key=lambda row: (row[2], row[3]))
IndexError: list index out of range
```

Reply:
```python
>>> import os
>>> who = [w.split() for w in os.popen('who').read().strip().split('\n')]
>>> who
[['awahab', 'pts/1', '2022-11-23', '17:56', '(IP REDACTED)']]
[['awahab', 'pts/0', '2022-11-23', '17:53', '(IP REDACTED)']]
>>> who.sort(key=lambda row: (row[2], row[3]))  # prioritize sort by date (row[2]) then time (row[3])
>>> who
[['awahab', 'pts/0', '2022-11-23', '17:53', '(IP REDACTED)']]
[['awahab', 'pts/1', '2022-11-23', '17:56', '(IP REDACTED)']]
>>> who[0][0]  # pick username of lowest index
'awahab'
```

Reply:
it appears there is inconsistency between your OS and ChimeraOS here

```
>>> import os
>>> who = [w.split() for w in os.popen('who').read().strip().split('\n')]
>>> who
[['gamer', '2022-11-23', '14:38'], ['gamer', 'pts/0', '2022-11-23', '14:41', '(REDACTED)']]
>>> who.sort(key=lambda row: (row[2], row[3]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <lambda>
IndexError: list index out of range
>>> 
```

Reply:
That's bizarre.  I tested with Python 3.8 and 3.10 on Ubuntu and Arch, respectively.  What version of Python 3 is Chimera using?
How about `who = sorted(who, key=lambda row: (row[2], row[3]))` ?

Reply:
Its not a python issue. `'who` in bash produces the bad output.

```
[gamer@chimeraos ~]$ who
gamer                 2022-11-23 14:15
gamer    pts/0        2022-11-23 16:21 (IP.ADD.RES.S)
```

Reply:
I believe the bottom entry is my SSH session and the top is the default session.

Reply:
Of course!  Sorry, was reading on a small screen and missed a column or two.  Just pushed a revised approach.